### DISCUSSION TITLE
Request body schema is interpreted incorrectly when using PEP 695-style with Pydantic `2.10`

### DISCUSSION THREAD

**Discussion by micuda** [fastapi/fastapi#12977] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# main.py

from typing import Annotated

from fastapi import FastAPI
from pydantic import BaseModel, Field
from typing_extensions import TypeAliasType

name_field = TypeAliasType(
    "name_field",
    Annotated[
        str,
        Field(default=None),
    ],
)


class PartialPerson(BaseModel):
    name: name_field


app = FastAPI()


@app.patch("/")
def update_person(data_to_update: PartialPerson) -> None:
    pass  # process partial update


# tests.py

import pytest
from fastapi.testclient import TestClient


@pytest.fixture
def api_client():
    return TestClient(app)


def test_update_person(api_client):
    response = api_client.patch("", json={})

    assert response.status_code == 200
```

### Description

When using Pydantic `2.10` the tests above fails with `422` status code (field `name` is required/missing). With Pydantic `2.9.2` and lower it passes as expected.

At the same time I'm able to create a valid instance of `PartialPerson` without provided value for `name` field (ie. `PartialPerson()`), which leads me it's issue of FastAPI instead of Pydantic itself.

Also I've found "similar" question/issue (https://github.com/fastapi/fastapi/discussions/10662), but related to dependencies and from it's description I found that it happens also with older versions of Pydantic, so I've decided to create new question, I hope it's OK. =)

### Operating System

Linux

### FastAPI Version

0.115.5

### Pydantic Version

2.10.0

### Python Version

3.11.2

**Comment by YuriiMotov** [comment]
Works fine with the current version of Pydantic (2.11.4)
